File: NUnit/Version2/Saucery2/Tests/SauceryIOSBase.cs
using System;
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Remote;
using Saucery2.Driver;
using Saucery2.Util;

namespace Saucery2.Tests {
    [TestFixture]
    public class SauceryIOSBase : SauceryRoot {
        protected SauceryIOSDriver Driver;

        public override void InitialiseDriver(DesiredCapabilities caps, int waitSecs) {
            try {
                Driver = new SauceryIOSDriver(caps);
                Driver.Manage().Timeouts().ImplicitlyWait(TimeSpan.FromSeconds(waitSecs));
            } catch(Exception ex) {
                Console.WriteLine(ex.Message);
            }
        }

        [TearDown]
        public void Cleanup() {
            if(Driver != null) {
                var passed = TestContext.CurrentContext.Result.Status == TestStatus.Passed;
                // log the result to SauceLabs
                SauceLabsStatusNotifier.NotifyStatus(Driver.GetSessionId(), passed);
                PrintSessionDetails();
                Driver.Quit();
            }
        }

        public void PrintSessionDetails() {
            try {
                var sessionId = Driver.GetSessionId();
                Console.WriteLine(@"SauceOnDemandSessionID={0} job-name={1}", sessionId, TestName);
            } catch(WebDriverException) {
                Console.WriteLine(@"Caught WebDriverException, quitting driver.");
                Driver.Quit();
            }
        }
    }
}
/*
 * Copyright Andrew Gray, Full Circle Solutions
 * Date: 12th July 2014
 * 
 */

File: JUnit/saucery-core/src/main/java/au/com/fullcirclesolutions/saucery/tests/SauceryIOSBase.java
package au.com.fullcirclesolutions.saucery.tests;

import au.com.fullcirclesolutions.saucery.driver.SauceryIOSDriver;
import au.com.fullcirclesolutions.saucery.utils.SauceryConstants;
import com.saucelabs.common.SauceOnDemandSessionIdProvider;
import com.saucelabs.junit.ConcurrentParameterized;
import java.net.URL;
import java.util.concurrent.TimeUnit;
import org.junit.After;
import org.junit.runner.RunWith;
import org.openqa.selenium.remote.DesiredCapabilities;

@RunWith(ConcurrentParameterized.class)
public class SauceryIOSBase extends SauceryRoot implements SauceOnDemandSessionIdProvider {
    protected SauceryIOSDriver Driver;

    public SauceryIOSBase(String os, String platformType, String browser, String browserVersion, String longName, String longVersion, String url, String device, String deviceOrientation) {
        super(os, platformType, browser, browserVersion, longName, longVersion, url, device, deviceOrientation);
    }

    @Override
    public void InitialiseDriver(DesiredCapabilities caps, int waitSecs) {
        SauceLabsFlowController.ControlFlow();
        try {
            String fullUrl = String.format(SauceryConstants.SAUCELABS_HUB, Authentication.getUsername(), Authentication.getAccessKey());
            Driver = new SauceryIOSDriver(new URL(fullUrl), caps);
            sessionId = Driver.getSessionId().toString();
            Driver.manage().timeouts().implicitlyWait(waitSecs, TimeUnit.SECONDS);
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
    }

    @After
    public void tearDown() {
        if (Driver != null) {
            // terminate the remote webdriver session
            Driver.quit();
        }
    }
}
/*
 * Copyright: Andrew Gray, Full Circle Solutions
 * Date: 12th July 2014
 */